### PR TITLE
Late multicellular musculature dependant speed

### PR DESCRIPTION
**Pull request by CI09**
**Brief Description of What This PR Does**

It isn't really necessary right now, but it makes late multicellular organisms' speed dependant on amount of muscular tissue (cell of that tissue has at least one myofibril). It might be fun addition until development focus switches to the stage itself.

I made a poll on the forums: https://community.revolutionarygamesstudio.com/t/thrive-development-polls/2971/34

Thing to note: there is no speed limit at the moment, so the organisms can swim/walk pretty fast.

**Related Issues**

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/late_multicellular_stage/LateMulticellularSpecies.cs [hunk 1]
```diff
@@ -190,6 +194,23 @@ private static float CalculateBrainPowerFromLayout(MetaballLayout<MulticellularM
...
 
+    private static float CalculateMuscularPowerFromLayout(MetaballLayout<MulticellularMetaball> layout, float scale)
+    {
+        float result = 0;
+
+        foreach (var metaball in layout)
+        {
+            if (metaball.CellType.IsMuscularTissueType())
+            {
+                // TODO: check that volume scaling in physically sensible way (using GetVolume) is what we want here
```
Comment (hhyyrylainen) [inline src/late_multicellular_stage/LateMulticellularSpecies.cs:205]: This comment is copy-pasted without adjustment from the brain power calculation...

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
I have just one small code comment. Does the speed get way too much with a bunch of muscle cells? If there is a chance that someone adds too much muscle and can't effectively pick up the items needed to advance, then a speed cap is kind of necessary.

**Comment by CI09** [comment]
Shall I use `Mathf.Clamp()`?

**Comment by hhyyrylainen** [comment]
Having a max speed would be excellent for making sure the game cannot become unplayable.
Just don't make the default speed too much slower to be annoying.

**Comment by CI09** [comment]
Also, I realized that organism's `Mass` could be utilized to slow down the movement, so muscles would have bigger role. Not necessary, but still a nice addition (I think).

**Comment by hhyyrylainen** [comment]
I wouldn't overcomplicate things with that. See for example what I needed to do to convert the mass based microbe approach to density based (to work with the Jolt physics engine), as that's what will need to happen to the prototypes as well when they are no longer prototypes.

**Pull request merged**